datalad/core/distributed/clone.py

```diff
@@ -585,15 +585,125 @@ def postclonecfg_ria(ds, props):
                    if s.get('annex-externaltype', None) == 'ora'
     ]
     if not ria_remotes:
-        lgr.debug("Found no RIA special remote")
+        # nothing autoenabled. Let's check origin's config for
+        # datalad.ora-remote.uuid as stored by create-sibling-ria and enable try
+        # enabling that one.
+        lgr.debug("Found no autoenabled ORA special remote. Trying to look it"
+                  "up in source config ...")
+
+        # First figure whether we cloned via SSH, HTTP or local path and then
+        # get that config file the same way:
+
+        config_content = None
+        scheme = props['source'].split(':')[0][4:]
+        if scheme == 'http':
+
+            # Note: Following outcommented code was proof how to check via HTTP.
+            #       However, for now we only want to annexenable origin in case
+            #       of HTTP, not the special remote. Hence it's uuid isn't
+            #       actually required. (see gh-4410)
+
+            # import requests
+            # try:
+            #     response = requests.get("{}{}config".format(
+            #         props['giturl'],
+            #         '/' if not props['giturl'].endswith('/') else '')
+            #     )
+            #     config_content = response.text
+            # except requests.RequestException as e:
+            #     lgr.debug("Failed to get config file from source:\n%s",
+            #               exc_str(e))
+
+            # config_content still None should lead to skip everything special
+            # remote related. Instead enable origin to deliver annex keys via
+            # HTTP:
+            lgr.debug("Trying to enable remote origin for git-annex")
+            ds.repo.enable_remote('origin')
+            ds.repo.fsck(remote='origin', fast=True)
+
+        elif scheme == 'ssh':
+
+            # TODO: switch the following to proper command abstraction:
+            from datalad.distributed.ora_remote import SSHRemoteIO, RIARemoteError
+            # SSHRemoteIO ignores the path part ATM. No remote CWD! (To be
+            # changed with command abstractions). So we need to get that part to
+            # have a valid path to origin's config file:
+            from datalad.support.network import URL
+            from datalad.utils import PurePosixPath
+            cfg_path = PurePosixPath(URL(props['giturl']).path) / 'config'
+            op = SSHRemoteIO(props['giturl'])
+            try:
+                # TODO: double-check path
+                response = op.read_file(cfg_path)
+                config_content = response
```

Comment:
why keep `response`?

Reply:
This can go indeed. Separation was an intermediate need.